core/sr-primitives/src/generic/unchecked_extrinsic.rs

```diff
@@ -123,6 +130,67 @@ where
 	}
 }
 
+/// A payload that has been signed for a unchecked extrinsics.
+///
+/// Note that the payload that we sign to produce unchecked extrinsic signature
+/// is going to be different than the `SignaturePayload` - so the thing the extrinsic
+/// actually contains.
+pub struct SignedPayload<Call, Extra: SignedExtension> {
+	raw_payload: (
+		Call,
+		Extra,
+		Extra::AdditionalSigned,
+	),
+}
+
+impl<Call, Extra> SignedPayload<Call, Extra> where
+	Call: Encode,
+	Extra: SignedExtension,
+{
+	/// Create new `SignedPayload`.
+	///
+	/// This function may fail if `additional_signed` of `Extra` is not available.
+	pub fn new(call: Call, extra: Extra) -> Result<Self, &'static str> {
+		let additional_signed = extra.additional_signed()?;
+		let raw_payload = (call, extra, additional_signed);
+		Ok(Self { raw_payload })
+	}
+
+	/// Create new `SignedPayload` from raw components.
+	pub fn from_raw(call: Call, extra: Extra, additional_signed: Extra::AdditionalSigned) -> Self {
+		Self {
+			raw_payload: (call, extra, additional_signed),
+		}
+	}
+
+	/// Get an encoded version of this payload.
+	///
+	/// Payloads longer than 256 bytes are going to be `blake2_256`-hashed.
+	pub fn using_encoded<O>(&self, f: impl FnOnce(&[u8]) -> O) -> O {
+		self.raw_payload.using_encoded(|payload| {
+			if payload.len() > 256 {
+				f(&blake2_256(payload)[..])
+			} else {
+				f(payload)
+			}
+		})
+	}
+
+	/// Deconstruct the payload into it's components.
+	pub fn deconstruct(self) -> (Call, Extra, Extra::AdditionalSigned) {
+		self.raw_payload
+	}
+}
+
+impl<Call, Extra> Encode for SignedPayload<Call, Extra> where
```

Comment:
Why is this implementation required?

Reply:
The generic `Signer` is working on anything that is `Encode`. I'll rather remove the `using_encoded` function, I've only added it to be able to document the behavior.

Reply:
You can also put documentation on top of a trait implementation. It will be shown in rustdocs as welll.